### PR TITLE
feat: add `@fig/autocomplete-generators` package

### PR DESCRIPTION
**Pull request by fedeci**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

**Comment by fedeci** [comment]
Ready!

File: packages/autocomplete-generators/src/filepaths.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  return {
+    template: "filepaths",
+    filterTemplateSuggestions: (suggestions = []) => {
+      const filtered = suggestions.filter(({ name = "", type }) => {
+        if (suggestFolders !== "filter" && type === "folder") {
+          return suggestFolders === "always";
+        }
+        if (equalsSet.has(name)) return true;
+        // We need to explicitly create a new Regexp object here or it will not work. Why?
+        if (matches && new RegExp(matches.source, matches.flags).test(name)) return true;
```
Comment (fedeci) [inline packages/autocomplete-generators/src/filepaths.ts:58]: Just for reference this is what I posted on the discord

File: packages/autocomplete-generators/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@withfig/autocomplete-generators",
```
Comment (fedeci) [inline packages/autocomplete-generators/package.json:2]:
```suggestion
  "name": "@fig/autocomplete-generators",
```

File: packages/autocomplete-generators/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# @withfig/autocomplete-generators package
```
Comment (fedeci) [inline packages/autocomplete-generators/README.md:1]:
```suggestion
# @fig/autocomplete-generators package
```

**Review by mschrage — APPROVED** [review]

**Pull request merged**